angr native compilation failure: sim_unicorn.cpp:1192:17: error: ‘out_of_range’ in namespace ‘std’ does not name a type

**Describe the bug.**

`angr_native` fails to compile on a current (2020-11-16) Arch Linux system. This still works on a Ubuntu System though, so this is potentially related to the compiler version or other system state.

**Environment Information.**

Clean python env. Unclear what global system state affects this.

```sh
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --with-isl --with-linker-hash-style=gnu --with-system-zlib --enable-__cxa_atexit --enable-cet=auto --enable-checking=release --enable-clocale=gnu --enable-default-pie --enable-default-ssp --enable-gnu-indirect-function --enable-gnu-unique-object --enable-install-libiberty --enable-linker-build-id --enable-lto --enable-multilib --enable-plugin --enable-shared --enable-threads=posix --disable-libssp --disable-libstdcxx-pch --disable-libunwind-exceptions --disable-werror gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.0 (GCC)
```

**To Reproduce.**

```sh
vf tmp
./extremely-simple-setup.sh
```

```
Running setup.py develop for angr
    ERROR: Command errored out with exit status 1:
     command: /home/fmagin/.virtualenvs/tempenv-093933134096/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/fmagin/Projects/angr-dev/angr/setup.py'"'"'; __file__='"'"'/home/fmagin/Projects/angr-dev/angr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /home/fmagin/Projects/angr-dev/angr/
    Complete output (11 lines):
    running develop
    Building angr_native
    g++ -O3 -fPIC -std=c++11 -I "/home/fmagin/.virtualenvs/tempenv-093933134096/lib/python3.8/site-packages/unicorn/include" -I "/home/fmagin/Projects/angr-dev/pyvex/pyvex/include" -o angr_native.so log.o sim_unicorn.cpp -lunicorn -lpyvex -L "/home/fmagin/.virtualenvs/tempenv-093933134096/lib/python3.8/site-packages/unicorn/lib" -L "/home/fmagin/Projects/angr-dev/pyvex/pyvex/lib" -shared
    sim_unicorn.cpp: In member function ‘taint_status_result_t State::get_final_taint_status(const std::unordered_set<taint_entity_t>&) const’:
    sim_unicorn.cpp:1192:17: error: ‘out_of_range’ in namespace ‘std’ does not name a type
     1192 |     catch (std::out_of_range) {
          |                 ^~~~~~~~~~~~
    make: *** [Makefile:37: angr_native.so] Error 1
    /home/fmagin/.virtualenvs/tempenv-093933134096/lib/python3.8/site-packages/setuptools/dist.py:466: UserWarning: The version specified ('9.0.gitrolling') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
      warnings.warn(
    error: Unable to build angr_native
    ----------------------------------------
  Rolling back uninstall of angr
  Moving to /home/fmagin/.virtualenvs/tempenv-093933134096/lib/python3.8/site-packages/angr.egg-link
   from /tmp/pip-uninstall-vem4zv7q/angr.egg-link
ERROR: Command errored out with exit status 1: /home/fmagin/.virtualenvs/tempenv-093933134096/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/fmagin/Projects/angr-dev/angr/setup.py'"'"'; __file__='"'"'/home/fmagin/Projects/angr-dev/angr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```

**Additional context.**

This is caused by @dnivra changes to unicorn in 01654ce02f85d44126d65bc25921f46efc8e3acc this might be something fairly trivial or obvious, but I don't really know anything about C++.

--- a/native/sim_unicorn.cpp
+++ b/native/sim_unicorn.cpp
@@ -8,6 +8,7 @@
 #include <memory>
 #include <map>
 #include <set>
+#include <stdexcept>
 #include <unordered_set>
 #include <unordered_map>
 #include <vector>

